src/lib/lockfile.ts

```diff
@@ -0,0 +1,34 @@
+import * as fs from 'fs';
+import { dirname } from 'path';
+
+import { exec, unlinkAll } from './fs-utils';
+
+// Equivalent to `drwxrwxrwt`
+const STICKY_WRITE_PERMISSIONS = parseInt('1777', 8);
+export const NOBODY_UID = 65534;
+
+export async function lock(path: string) {
```

Comment:
Something like this would make this function a bit more generic.

```suggestion
export async function lock(path: string, uid = NOBODY_UID) {
```